lib/queue.js

```diff
@@ -243,12 +243,9 @@ Queue.prototype.checkHealth = function (cb) {
   return promise;
 };
 
-Queue.prototype._scanForJobs = function (cursor, set, cb) {
-  // Ideally, we would be able to paginate redis sets like we do sorted ranges.
-  // For an MVP, however, we're aiming to replace our usage of Arena for Bull,
-  // and furthermore Bull 1.x calls SMEMBERS -- which is more blocking than SSCAN.
-
-  this.client.sscan(this.toKey('failed'), cursor, 'COUNT', this.settings.redisScanCount, (err, results) => {
+Queue.prototype._scanForJobs = function (key, cursor, size, set, cb) {
```

Comment:
It's a little strange that this accepts a set as an input, but produces an array. I'd be tempted to have the caller convert to an array instead.